robot-name/robot-name_test.spec.js

```diff
@@ -12,9 +12,17 @@ describe("Robot", function() {
   });
 
   it("different robots have different names", function() {
-    var robotOne = new Robot();
-    var robotTwo = new Robot();
-    expect(robotOne.name).not.toEqual(robotTwo.name);
+    var usedNames= {};
+
+    var i = 0,
+        max = 10000;
+
+    for (i; i < max; i++) {
```

Comment:
Based on what I've seen on SO, eloquentjavascript.net, etc., it is not common. Personally, I also prefer:

    for (var i = 0; i < 10000; i++) {

The reasons I declared outside the loop are:

1) [A style guide](https://github.com/rwaldron/idiomatic.js/) suggests this way. Ctrl+f: "length = 100;"

2) A JSLint tool I ran wanted them declared outside.

3) Apparently it is [the "JSLint" way](http://stackoverflow.com/questions/4646455/jslint-error-move-all-var-declarations-to-the-top-of-the-function), the controversial gold standard for JS. But I agree with the top/accepted answer to the linked SO. Even though declaring outside makes academic sense...blech, I'm abandoning it.